AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Service/Storage/AWSS3StorageService.swift

```diff
@@ -21,7 +22,12 @@ class AWSS3StorageService: AWSS3StorageServiceBehaviour, StorageServiceProxy {
     var region: String!
     var bucket: String!
 
-    var s3Client: S3Client!
+    /// - Tag: AWSS3StorageService.s3Client
+    @available(*, deprecated, renamed: "client")
+    var s3Client: S3Client { didSet { client = s3Client } }
+
+    /// - Tag: AWSS3StorageService.client
+    var client: S3ClientProtocol
```

Comment:
Using the protocol to make unit testing more practical. Sadly, the concrete class cannot be removed because the public API for the escape hatch uses the class rather than the protocol.